### ISSUE TITLE
Changing color on settings page using hex input doesn't update it immediately

### ISSUE THREAD

**Issue by mxmou** [ScratchAddons/ScratchAddons#2096]
**Describe the bug**

Using the hex input to change a color on the settings page doesn't update the color until the color picker is closed and reopened.

**To Reproduce**

1. Open the settings page
2. Open a color picker
3. Type a valid color into the hex input
4. Close the picker
5. The color doesn't update
6. Open the picker again
7. The color updates

**Expected behavior**

The color would update immediately or at least when closing the picker.

**Comment by WorldLanguages** [comment]
smh this color picker is causing so many problems for transparency's sake lmao

**Comment by mxmou** [comment]
> smh this color picker is causing so many problems for transparency's sake lmao

Maybe we could use a normal `<input type="color" />` and make transparency a separate input.

**Comment by GrahamSH-LLK** [comment]
@mxmou Can you reproduce this if you click out of the input?

**Comment by mxmou** [comment]
@GrahamSH-LLK No, this only happens if the picker is closed immediately after changing the input value.

**Comment by GrahamSH-LLK** [comment]
> @GrahamSH-LLK No, this only happens if the picker is closed immediately after changing the input value.

So, I _think_ what I need to do is de-focus the input on close.

**Comment by GrahamSH-LLK** [comment]
Actually, I think executing _valueChanged on close will work. I'll test when I get access to my dev env.

**Comment by RIMOPA** [comment]
I made a specification:
1. When I put a color (writing it) it doesn't charge until I press enter, if I don't press it, It doesn't update until I click that selector again.

2. When the color gets updated, the XY(of the 2D selector), X (of the slider) and the circle preview gets updated but the value of the slider doesn't update the 2D selector (the X and Y gets updated, but the color don't)(see image in thread)

3. When the text has the "#" the color is 000000 instead of deleting the "#".

**Comment by mxmou** [comment]
> When the text has the "#" the color is 000000 instead of deleting the "#".

That should be reported separately.
